### ISSUE TITLE
handleViewException only accepts Exceptions

### ISSUE THREAD

**Issue by bofru** [livewire/livewire#612]
**Describe the bug**
If an error occurs on a view, it gets passed through the method '[handleViewException](https://github.com/livewire/livewire/blob/871d0ca3a1a2ad70f7caa96b4e350d72d72798ec/src/CompilerEngine.php#L47)' in the CompilerEngine.php file. This method only accepts objects of the type 'Exception' although other types of Errors may occur (e.g. ParseError).
E.g. if a ParseError occurs, the developer does not see the actual ParseError but instead gets an Exception that says that the passed ParseError Object does not match the expected Exception type (`Argument 1 passed to Livewire\CompilerEngine::handleViewException() must be an instance of Exception, instance of ParseError given, [...] `)

**To Reproduce**

- 1.a Create a test component and edit the blade to provoke an ParseError:

```
<div>
    @if($foobar

    @endif
</div>
```

Output:
`Argument 1 passed to Illuminate\View\Engines\CompilerEngine::handleViewException() must be an instance of Exception, instance of ParseError given, called in [...]`

- 1.b alternatively provoke an Error by using Auth::user()->can('foobar') on null when not being logged in:

```
<div>
    @if(Auth::user()->can('foobar'))

    @endif
</div>
```

Output:
`Argument 1 passed to Illuminate\View\Engines\CompilerEngine::handleViewException() must be an instance of Exception, instance of Error given, called in  [...]`

**Expected behavior**
I think one should get the actual error/exception (?), in this case:
- 1.a `syntax error, unexpected ':', expecting '('  [...] `
- 1.b `Call to a member function can() on null [...]`
Otherwise it may be a little tricky to find the real cause of the error.

**Desktop**
 - laravel/framework v6.14.0
 - livewire/livewire v0.7.4

**Comment by calebporzio** [comment]
Thanks for the good issue description.

Is the solution to catch `Throwable` instead?

**Comment by bofru** [comment]
Catching `Throwable` solves the problem for me, however just changing the type to `Throwable` will not work as Laravel 6.x expects an `Exception` in the [`handleViewExceptionMethod`](https://laravel.com/api/6.x/Illuminate/View/Engines/PhpEngine.html#method_handleViewException). Locally, I have fixed this by checking wether the `Throwable` is an instance of `Exception` or not. If not, I simply throw it with `throw($e)`. I have opened a pull request that could be merged if this approach is appropriate.
